### PR TITLE
fix issue #103 - reorder version dumping

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**

### CODE REVIEW

File: setuptools_scm/__init__.py [hunk 1]
```diff
@@ -37,6 +37,7 @@ def _version_from_entrypoint(root, entrypoint):
 
 
 def dump_version(root, version, write_to, template=None):
+    assert isinstance(version, string_types)
```
Comment (anarcat) [inline setuptools_scm/__init__.py:41]: does this detect the new failure?

File: setuptools_scm/__init__.py [hunk 2]
```diff
@@ -101,18 +102,20 @@ def get_version(root='.',
...
             template=write_to_template)
-        if isinstance(version, string_types):
-            return version
-        version = format_version(
-            version,
-            version_scheme=version_scheme,
-            local_scheme=local_scheme)
-        return version
+
+        return version_string
```
Comment (anarcat) [inline setuptools_scm/__init__.py:130]: i like this: much clearer than the previous implementation - i thought of doing something like that in my own PR but refrained to keep things simple...

File: testing/test_functions.py [hunk 3]
```diff
@@ -52,7 +52,7 @@ def test_format_version(version, monkeypatch, scheme, expected):
 
 def test_dump_version_doesnt_bail_on_value_error(tmpdir):
     write_to = "VERSION"
-    version = VERSIONS['exact']
+    version = str(VERSIONS['exact'].tag)
```
Comment (anarcat) [inline testing/test_functions.py:55]: not sure what isthis for - would this silence possible unit test failures that could detect the fixed bug?

**Review by anarcat — COMMENTED** [review]
looks good, but i'm wondering if the test suite would find a similar regression in the future.

File: testing/test_functions.py (hunk 3 above)
Comment (RonnyPfannschmidt) [reply testing/test_functions.py:55]: exact.tag is a setultools version not a string

File: setuptools_scm/__init__.py (hunk 2 above)
Comment (RonnyPfannschmidt) [reply setuptools_scm/__init__.py:130]: i did a more intensive restructure ensuring i never get a string there - the shortcut was an premature optimization, and as such the root of the evil ^^

File: setuptools_scm/__init__.py (hunk 1 above)
Comment (RonnyPfannschmidt) [reply setuptools_scm/__init__.py:41]: the code in dump-version only works with strings, perhaps even this check is not trough enough

**Comment by RonnyPfannschmidt** [comment]
@anarcat @untitaker unless you have further comments i'm about to release 1.13

**Comment by untitaker** [comment]
lgtm

**Pull request merged**